project/products/pycharm/src/net/ashald/envfile/products/pycharm/PyCharmRunConfigurationExtension.java

```diff
@@ -50,7 +50,7 @@ protected void validateConfiguration(@NotNull AbstractPythonRunConfiguration con
 
     @Override
     protected void patchCommandLine(@NotNull AbstractPythonRunConfiguration configuration, @Nullable RunnerSettings runnerSettings, @NotNull GeneralCommandLine cmdLine, @NotNull String runnerId) throws ExecutionException {
-        EnvFileConfigurationEditor.patchEnvironmentVariables(configuration, cmdLine.getEnvironment());
+        EnvFileConfigurationEditor.patchEnv(configuration, cmdLine.getEnvironment());
```

Comment:
Here and in RubyMine integration below let's do:
```
Map<String, String> cmdEnvironment = cmdLine.getEnvironment();
Map<String, String> newEnv = EnvFileConfigurationEditor.collectEnv(configuration, cmdEnvironment);
cmdEnvironment.clear();
cmdEnvironment.putAll(newEnv);
```

Even though it's more code right away, we're maintaining a pure interface with the plugin core and containing all the specifics of platform integration within respective extension files.

Reply:
At first I did it exactly that way but in order to get rid of the code duplication I restored the `patchEnv` method. 
I reverted the last commit with the `patchEnv` method:)